zerver/views/streams.py

```diff
@@ -369,6 +369,19 @@ def add_subscriptions_backend(
     (subscribed, already_subscribed) = bulk_add_subscriptions(streams, subscribers,
                                                               acting_user=user_profile, color_map=color_map)
 
+    # Make acting user a stream admin, if acting user subscribed to newly created streams.
+    if user_profile.email in principals:
+        for stream in created_streams:
+            recipient = stream.recipient
+            try:
+                sub = Subscription.objects.get(user_profile=user_profile,
+                                               recipient=recipient)
+            except Subscription.DoesNotExist:  # nocoverage
+                # If user is present in principals, then user should be subscribed to stream. But
+                # in case if not then just pass.
+                pass
+            do_change_subscription_property(user_profile, sub, stream, "is_stream_admin", True)
```

Comment:
This is not correct; we should instead make `bulk_add_subscriptions` be responsible for setting the property at the moment is creates the initial `Subscription` row for the user if they just created the stream.  

(There are other code paths that could call `bulk_add_subscriptions` after creating a stream)

Reply:
I think we need to pass created streams separately to bulk_add_subscriptions, to set stream admin property for newly created stream only. I am unable to find any other way of checking if the stream is just created or not.

Reply:
I think we need to pass created streams separately to bulk_add_subscriptions, to set stream admin property for newly created stream only. I am unable to find any other way of checking if the stream is just created or not.

Reply:
I am not sure but I think passing created streams and existed streams separately might make the code look ugly, as in some cases we call `bulk_add_subscriptions` only for newly created streams and in some case we pass both exisiting and newly created streams.